crates/router/src/services/authentication.rs

```diff
@@ -683,6 +684,78 @@ where
     }
 }
 
+#[derive(Debug, Default)]
+pub struct AdminApiAuthWithMerchantId(AdminApiAuth);
+
+#[async_trait]
+impl<A> AuthenticateAndFetch<AuthenticationData, A> for AdminApiAuthWithMerchantId
+where
+    A: SessionStateInfo + Sync,
+{
+    async fn authenticate_and_fetch(
+        &self,
+        request_headers: &HeaderMap,
+        state: &A,
+    ) -> RouterResult<(AuthenticationData, AuthenticationType)> {
+        let (_, authentication_type) = self
+            .0
+            .authenticate_and_fetch(request_headers, state)
+            .await?;
```

Comment:
The authentication_type for this type should not be same one as `AdminApiKey`. Please construct a authentication_type separately which has merchant_id.

If the existing variants are not suitable, you can create something like `AdminApiAuthWithMerchantId { merchant_id: String }`